crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -951,10 +949,46 @@ impl EntityCommands<'_> {
     /// # bevy_ecs::system::assert_is_system(add_combat_stats_system);
     /// ```
     #[track_caller]
-    pub fn insert(&mut self, bundle: impl Bundle) -> &mut Self {
+    pub fn insert(self, bundle: impl Bundle) -> Self {
         self.add(insert(bundle, InsertMode::Replace))
     }
 
+    /// Similar to [`Self::insert`] but will only insert if the predicate returns true.
+    ///
+    /// # Panics
+    ///
+    /// The command will panic when applied if the associated entity does not exist.
+    ///
+    /// To avoid a panic in this case, use the command [`Self::try_insert_if`] instead.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_ecs::prelude::*;
+    /// # #[derive(Resource)]
+    /// # struct PlayerEntity { entity: Entity }
+    /// # impl PlayerEntity { fn is_spawned(&self) -> bool { true } }
+    /// #[derive(Component)]
+    /// struct Health(u32);
+    ///
+    /// fn add_health_system(mut commands: Commands, player: Res<PlayerEntity>) {
+    ///     commands.entity(player.entity)
+    ///         .insert_if(Health(10), || player.is_spawned());
+    /// }
```

Comment:
```suggestion
    /// # #[derive(Component)]
    /// # struct StillLoadingStats;
    /// #[derive(Component)]
    /// struct Health(u32);
    ///
    /// fn add_health_system(mut commands: Commands, player: Res<PlayerEntity>) {
    ///     commands
    ///         .entity(player.entity)
    ///         .insert_if(Health(10), || player.is_spawned())
    ///         .remove::<StillLoadingStats>();
    /// }
```
This comment is also valid for the other methods

Reply:
Are we sure this is working the way we think it is? I would think for that example you'd only want to remove `StillLoadingStats` if `Health` was successfully inserted, which you can't actually infer here? Am I misreading the intent on this?

Reply:
Oh right, good catch. It would make more sense if we swapped `player.is_spawned()` with `!player.is_spectator()`. Or another example, whatever you like :) I just want to make sure we show that this is cool for chaining.